src/librustc/middle/cstore.rs

```diff
@@ -100,18 +100,100 @@ pub struct NativeLibrary {
 /// part of the AST that we parse from a file, but it becomes part of the tree
 /// that we trans.
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
-pub enum InlinedItem {
-    Item(DefId /* def-id in source crate */, P<hir::Item>),
-    TraitItem(DefId /* impl id */, P<hir::TraitItem>),
-    ImplItem(DefId /* impl id */, P<hir::ImplItem>)
+pub struct InlinedItem {
```

Comment:
Seems okay - shouldn't need the type of a constant though, in fact, you should be able to only record a `Vec<DefId>` for the arguments of a `const fn`, which doesn't even need to be in a separate enum anymore, just replace the `kind` field with `const_fn_args: Vec<DefId>` and that should be that.